gui/wxpython/gui_core/mapdisp.py

```diff
@@ -160,17 +160,29 @@ def GetToolSwitcher(self):
 
     def SetProperty(self, name, value):
         """Sets property"""
-        if self.HasProperty("projection"):
-            self.statusbarManager.SetProperty(name, value)
-        else:
+        if name == "projection":
             self.mapWindowProperties.useDefinedProjection = value
+        elif name == "resolution":
+            self.mapWindowProperties.resolution = value
+        elif name == "alignExtent":
+            self.mapWindowProperties.alignExtent = value
+        elif name == "region":
+            self.mapWindowProperties.showRegion = value
+        else:
+            self.statusbarManager.SetProperty(name, value)
 
     def GetProperty(self, name):
         """Returns property"""
-        if self.HasProperty("projection"):
-            return self.statusbarManager.GetProperty(name)
-        else:
+        if name == "projection":
             return self.mapWindowProperties.useDefinedProjection
+        elif name == "resolution":
+            return self.mapWindowProperties.resolution
+        elif name == "alignExtent":
+            return self.mapWindowProperties.alignExtent
+        elif name == "region":
+            return self.mapWindowProperties.showRegion
+        else:
+            return self.statusbarManager.GetProperty(name)
 
```

Comment:
You could use hasattr/getattr here instead of those ifs.

Reply:
I have tried that in the last commit but a bug with useDefinedProjection apparently appeared. The "UseDefinedProjection" toggle button does not work. Now I am going to bed, I will look at it more deeply tomorrow.